src/libsync/propagatedownload.cpp

```diff
@@ -451,19 +436,76 @@ void PropagateDownloadFile::startDownload()
     {
         SyncJournalDb::DownloadInfo pi;
         pi._etag = _item->_etag;
-        pi._tmpfile = tmpFileName;
+        pi._tmpfile = _tmpFileName;
         pi._valid = true;
         propagator()->_journal->setDownloadInfo(_item->_file, pi);
         propagator()->_journal->commit("download file start");
     }
 
+    if (isZsyncPropagationEnabled(propagator(), _item)) {
+        QFile file(propagator()->getFilePath(_item->_file));
+        if (file.open(QIODevice::ReadOnly)) {
```

Comment:
This check seems odd, since file isn't used below. You could check `_previousSize > 0`? If you do want to keep it, produce an error `if (!file.open(...))` and de-indent the happy case.

Reply:
I changed it to a `QTemporaryFile`, this check is just to check that we have a file to seed from, otherwise no point doing delta-sync. I removed the else and removed some duplicate code since it can fall-through. Hard to do that if I de-indent the happy case, let me know your thoughts now.

Reply:
I didn't see the fallthrough, makes sense then. 

You shouldn't have to go to the filesystem to check whether there's a base file since the previous phases have already discovered all of that - hence I suggested just checking `_previousSize`. But it's a minor thing.

Reply:
checking _previousSize > 0 instead now, fixed.